openstf/pipeline/train_predict_backtest.py

```diff
@@ -0,0 +1,135 @@
+# SPDX-FileCopyrightText: 2017-2021 Alliander N.V. <korte.termijn.prognoses@alliander.com> # noqa E501>
+#
+# SPDX-License-Identifier: MPL-2.0
+from typing import Union, List
+from pathlib import Path
+import logging
+
+import pandas as pd
+
+from openstf.model.confidence_interval_applicator import ConfidenceIntervalApplicator
+from openstf.postprocessing.postprocessing import (
+    add_prediction_job_properties_to_forecast,
+)
+from openstf.model.standard_deviation_generator import StandardDeviationGenerator
+from openstf.model.model_creator import ModelCreator
+from openstf.metrics.reporter import Reporter
+from openstf.model.serializer import PersistentStorageSerializer
+from openstf.model_selection.model_selection import split_data_train_validation_test
+from openstf.validation import validation
+from openstf.feature_engineering.feature_applicator import TrainFeatureApplicator
+
+DEFAULT_TRAIN_HORIZONS: List[float] = [0.25, 24.0]
+DEFAULT_EARLY_STOPPING_ROUNDS: int = 10
+
+
+def train_model_and_forecast_back_test(
+    pj: dict,
+    input_data: pd.DataFrame,
+    trained_models_folder: Union[str, Path],
+    save_figures_folder: Union[str, Path],
+    training_horizons: List[float] = None,
+) -> pd.DataFrame:
+    """Pipeline for a back test.
+
+        DO NOT USE THIS PIPELINE FOR OPERATIONAL FORECASTS
+
+    Args:
+        pj (dict): Prediction job.
+        input_data (pd.DataFrame): Input data
+        trained_models_folder:
+        trained_models_folder (Path): Path where trained models are stored
+        save_figures_folder (Path): path were reports (mostly figures) about the training procces are stored
+        training_horizons (list): Horizons to train on in hours.
+            These horizons are also used to make predictions (one for every horizon)
+
+    Returns:
+        forecast (pandas.DataFrame)
+
+    """
+    if training_horizons is None:
```

Comment:
This part (line 50 to 102) looks identical to the train_model_core() function line 117:201. Could we not just use that? Or did I miss something perhaps?

Reply:
Yes, now made a train_pipeline_common function